airflow/cli/cli_parser.py

```diff
@@ -137,9 +137,12 @@ def get_long_option(arg: Arg):
 
 
 def _add_command(subparsers: argparse._SubParsersAction, sub: CLICommand) -> None:
-    sub_proc = subparsers.add_parser(
-        sub.name, help=sub.help, description=sub.description or sub.help, epilog=sub.epilog
-    )
+    if isinstance(sub, ActionCommand) and sub.hide:
+        sub_proc = subparsers.add_parser(sub.name, epilog=sub.epilog)
```

Comment:
How does this work if someone ran `airflow db upgrade --help`? Do they get nothing?

Reply:
Nope, they do get all the args (only help is hidden) --
![image](https://github.com/apache/airflow/assets/35839624/3eadeeb2-0e05-43f0-9f3f-2923f4dde74e)